service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/MhsPayloadMapper.java

```diff
@@ -0,0 +1,45 @@
+package uk.nhs.adaptors.gp2gp.gpc;
+
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.ChronoField;
+
+import lombok.RequiredArgsConstructor;
+import uk.nhs.adaptors.gp2gp.common.service.TimestampService;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+import uk.nhs.adaptors.gp2gp.mhs.model.MhsPayloadTemplateParameters;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.github.mustachejava.Mustache;
+
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Component
+public class MhsPayloadMapper {
```

Comment:
Class name is misleading. "MhsPayload" would implies a payload for the MHS Adaptor's API which this is not.

Maybe `EhrDocumentMapper`? Need to change other classes as well.